internal/controller/cloudstorage_controller.go

```diff
@@ -128,41 +139,91 @@ func (b CloudStorageReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if secretName, err = stsflow.STSStandardizedFlow(); err != nil {
 		logger.Error(err, "unable to get STS Secret")
-		b.EventRecorder.Event(&bucket, corev1.EventTypeWarning, "UnableToSTSSecret", fmt.Sprintf("unable to delete bucket: %v", bucket.Spec.Name))
-		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+		b.EventRecorder.Event(&bucket, corev1.EventTypeWarning, "UnableToSTSSecret", fmt.Sprintf("unable to get STS secret: %v", err))
+		// Set condition and return error to trigger exponential backoff
+		apimeta.SetStatusCondition(&bucket.Status.Conditions, metav1.Condition{
+			Type:    oadpv1alpha1.ConditionBucketReady,
+			Status:  metav1.ConditionFalse,
+			Reason:  oadpv1alpha1.ReasonSTSSecretError,
+			Message: fmt.Sprintf("Unable to get STS secret: %v", err),
+		})
+		bucket.Status.LastSynced = &metav1.Time{Time: time.Now()}
+		if updateErr := b.Client.Status().Update(ctx, &bucket); updateErr != nil {
+			logger.Error(updateErr, "failed to update CloudStorage status")
+		}
+		return ctrl.Result{}, err
 	}
 	if secretName != "" {
-		// Secret was created successfully by STSStandardizedFlow
-		logger.Info(fmt.Sprintf("Following standardized STS workflow, secret %s created successfully", secretName))
+		// Secret exists after STSStandardizedFlow (may have been created, updated, or unchanged)
+		logger.V(1).Info(fmt.Sprintf("Following standardized STS workflow, secret %s is available", secretName))
 	}
 	// Now continue with bucket creation as secret exists and we are good to go !!!
 	if ok, err = clnt.Exists(); !ok && err == nil {
-		// Handle Creation if not exist.
+		// Handle Creation if bucket does not exist
 		created, err := clnt.Create()
-		if !created {
-			logger.Info("unable to create object bucket")
+		if !created || err != nil {
+			logger.Info("unable to create object bucket", "error", err)
 			b.EventRecorder.Event(&bucket, corev1.EventTypeWarning, "BucketNotCreated", fmt.Sprintf("unable to create bucket: %v", err))
-			return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
-		}
-		if err != nil {
-			//TODO: LOG/EVENT THE MESSAGE
-			logger.Error(err, "Error while creating event")
-			return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+			// Set condition and return error to trigger exponential backoff
+			apimeta.SetStatusCondition(&bucket.Status.Conditions, metav1.Condition{
+				Type:    oadpv1alpha1.ConditionBucketReady,
+				Status:  metav1.ConditionFalse,
+				Reason:  oadpv1alpha1.ReasonBucketCreationFailed,
+				Message: fmt.Sprintf("Failed to create bucket: %v", err),
+			})
+			bucket.Status.LastSynced = &metav1.Time{Time: time.Now()}
+			bucket.Status.Name = bucket.Spec.Name
+			if updateErr := b.Client.Status().Update(ctx, &bucket); updateErr != nil {
+				logger.Error(updateErr, "failed to update CloudStorage status")
+			}
+			// Return error to trigger exponential backoff
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, fmt.Errorf("bucket creation failed")
```

Comment:
`return ctrl.Result{}, err` makes it requeue with exponential backoff if `err != nil`.